sebs/storage/resources.py

```
@@ -19,178 +19,259 @@
-            storage_impl = config["storage"][storage_type]["type"]
-            storage_config = config["storage"][storage_type][storage_impl]
-            self.logging.info(
-                "Using user-provided configuration of storage "
-                f"type: {storage_type} for {self._name} containers."
-            )
+    def _deserialize_storage_config( # Renamed for clarity, takes specific storage_type_key
+        self,
+        user_config: dict, # User-provided config for the entire 'resources' section
+        cached_resource_config: Optional[dict], # Cached 'resources' section
+        storage_type_key: str, # "object_storage" or "nosql_storage"
+        default_type_map: Dict[str, Type[PersistentStorageConfig] | Type[NoSQLStorageConfig]]
```

Ruff (0.11.9)

Undefined name `Dict`

(F821)